extensions/geode-modules/src/main/java/org/apache/geode/modules/session/catalina/DeltaSessionManager.java

```diff
@@ -932,16 +945,57 @@ private void doLoad() throws ClassNotFoundException, IOException {
    * Return a File object representing the pathname to our persistence file, if any.
    */
   private File sessionStore(String ctxPath) {
-    String storeDir = System.getProperty("catalina.base");
+    String storeDir = getSystemPropertyValue(catalinaBaseSystemProperty);
     if (storeDir == null || storeDir.isEmpty()) {
-      storeDir = System.getProperty("java.io.tmpdir");
+      storeDir = getSystemPropertyValue(javaTempDirSystemProperty);
     } else {
-      storeDir += System.getProperty("file.separator") + "temp";
+      storeDir += getSystemPropertyValue(fileSeparatorSystemProperty) + "temp";
     }
 
+    return getFileAtPath(storeDir, ctxPath);
+  }
+
+  String getSystemPropertyValue(String propertyKey) {
+    return System.getProperty(propertyKey);
+  }
+
+  File getFileAtPath(String storeDir, String ctxPath) {
     return (new File(storeDir, ctxPath.replaceAll("/", "_") + ".sessions.ser"));
   }
 
+  FileInputStream getFileInputStream(File file) throws FileNotFoundException {
+    return new FileInputStream(file.getAbsolutePath());
+  }
+
+  BufferedInputStream getBufferedInputStream(FileInputStream fis) {
+    return new BufferedInputStream(fis);
+  }
+
+  ObjectInputStream getObjectInputStream(BufferedInputStream bis) throws IOException {
+    return new ObjectInputStream(bis);
+  }
+
+  FileOutputStream getFileOutputStream(File file) throws FileNotFoundException {
+    return new FileOutputStream(file.getAbsolutePath());
+  }
+
+  BufferedOutputStream getBufferedOutputStream(FileOutputStream fos) {
+    return new BufferedOutputStream(fos);
+  }
+
+  ObjectOutputStream getObjectOutputStream(BufferedOutputStream bos) throws IOException {
+    return new ObjectOutputStream(bos);
+  }
+
+  void writeToObjectOutputStream(ObjectOutputStream oos, List listToWrite) throws IOException {
+    oos.writeObject(listToWrite.size());
+  }
+
+  int getSessionCountFromObjectInputStream(ObjectInputStream ois)
+      throws IOException, ClassNotFoundException {
+    return (Integer) ois.readObject();
+  }
+
```

Comment:
Do we really need all these new methods to create, write, and other operations on low level java classes just for mocking purposes?. Looks "a bit overkilling" to me...

Reply:
The problem is all these input streams are being created, used, and discarded internally in the load/unload process. This means that if we want to verify any behavior of those objects (in addition to avoiding the NullPointerExceptions that come with trying to actually perform these filesystem operations in this unit test environment) we have to mock them, and as far as I could tell the best way to do that here was to create these new methods so that the spy could use them to return mocks instead of freshly constructed objects. The writeObject method is also necessary because the call it's making internally is final and thus can't be mocked. This method throws exceptions if run on a mocked output stream so we need to call it in a helper method to prevent execution during testing.